### PR TITLE
new c.pest file

### PR DESCRIPTION
**Pull request by williamaguessy**
I tried to ad the possibility of having "multiligne string" on c.pest. It works but after the first string detection in the pest editor, the parser is not able to recognise the comment anymore.

### CODE REVIEW

**Comment by bjones1** [comment]
- [ ] Remove commits of `Cargo.lock` or `package.lock`.
- [ ] Send this PR to the `pest` branch, instead of the `main` branch.

**Comment by williamaguessy** [comment]
I put my change into the python pest file, for now it can well see inline comment in multiligne string as code block in this case ''' # this in not a comment ''', but not is this case : print(''' # this is not a comment '''). I aslo tried to revert my previous wrong commits.

File: server/src/lexer/pest_parser.rs [hunk 1]
```diff
@@ -231,7 +231,7 @@ mod test {
     use crate::lexer::{CodeDocBlock, DocBlock};
```
Comment (bjones1) [inline server/src/lexer/pest_parser.rs:231]: Revert these changes.

File: server/src/lexer/pest/python.pest [hunk 2]
```diff
@@ -54,34 +54,37 @@ inline_comment_contents = { logical_line ~ newline_eoi }
...
 // Use this so that the ending (usually a newline) gets captured in a token.
-block_comment_ending = { newline_eoi }
+// block_comment_ending = { newline_eoi }
 
 // ## Code blocks
-code_block = { code_line+ }
+
+multiline_string = { "'''" ~ (!"'''" ~ ANY)* ~ "'''" }
+root = _{ multiline_string | inline_comment }
+code_block = { multiline_string | code_line+ }
```
Comment (bjones1) [inline server/src/lexer/pest/python.pest:87]: Not the right approach -- a code block is:
- Not a doc block, followed by non-newline characters then a newline OR
- Not a doc block, followed by non-newline characters  then EOI OR
- non-newline characters followed by a newline OR
- a multi-line string

**Review by bjones1 — CHANGES_REQUESTED** [review]

**Comment by bjones1** [comment]
> I put my change into the python pest file, for now it can well see inline comment in multiligne string as code block in this case ''' # this in not a comment ''', but not is this case : print(''' # this is not a comment '''). I aslo tried to revert my previous wrong commits.

Progress -- see my review comments.

**Comment by bjones1** [comment]
I see progress, but nothing from what we discussed in my office. I want a fully-working PR ready before we meet tomorrow.

**Comment by williamaguessy** [comment]
Here is the change we discussed during our meeting but it's not working for the moment. It only recognise multiligne string but not normal code block anymore

**Comment by williamaguessy** [comment]
in fact it detect only the multiline string evne if it is one of the two other case. The definition isn't complete

**Comment by williamaguessy** [comment]
I have made the changes to handle single and double quotes with backslashes, and I have also added support for parsing both ''' and """ multiline strings. Initially, there was a conflict between the two rules because the parser interpreted the first two double quotes as the start of a multiline backslash. To resolve this, I replaced a * with a +. multiline_string_backslash = { 
"\"" ~ (("\\" ~ ANY) | (!( "\"" | NEWLINE ) ~ ANY))**+** ~ "\"" 
}

**Pull request closed without merging**